hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/TokenStateTranslator.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.mono.state.migration;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.Key;
+import com.hedera.hapi.node.base.TokenID;
+import com.hedera.hapi.node.base.TokenSupplyType;
+import com.hedera.hapi.node.base.TokenType;
+import com.hedera.hapi.node.state.token.Token;
+import com.hedera.hapi.node.transaction.CustomFee;
+import com.hedera.node.app.service.mono.legacy.core.jproto.JKey;
+import com.hedera.node.app.service.mono.pbj.PbjConverter;
+import com.hedera.node.app.service.mono.state.submerkle.EntityId;
+import com.hedera.node.app.service.mono.state.submerkle.FcCustomFee;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hedera.node.app.service.token.ReadableTokenStore;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Translates between the legacy {@link com.hedera.node.app.service.mono.state.merkle.MerkleToken} and the {@link Token} and vise versa.
+ */
+public class TokenStateTranslator {
+
+    /**
+     * Translates the {@link com.hedera.node.app.service.mono.state.merkle.MerkleToken} to the {@link Token}.
+     * @param token {@link com.hedera.node.app.service.mono.state.merkle.MerkleToken}
+     * @return  {@link Token}
+     */
+    public static Token tokenFromMerkle(
+            @NonNull final com.hedera.node.app.service.mono.state.merkle.MerkleToken token) {
+        final var builder = Token.newBuilder()
+                .tokenNumber(token.getKey().longValue())
+                .name(token.name())
+                .symbol(token.symbol())
+                .decimals(token.decimals())
+                .totalSupply(token.totalSupply())
+                .treasuryAccountNumber(token.treasury().num())
+                .lastUsedSerialNumber(token.getLastUsedSerialNumber())
+                .deleted(token.isDeleted())
+                .tokenType(fromMerkleType(token.tokenType()))
+                .supplyType(fromMerkleSupplyType(token.supplyType()))
+                .autoRenewAccountNumber(Optional.ofNullable(token.autoRenewAccount())
```

Comment:
This would be simpler (and avoid creating 5 objects) as follows:
```java
                .autoRenewAccountNumber(token.autoRenewAccount() != null
                        ? token.autoRenewAccount().num()
                        : 0L)
```

Reply:
done thanks